pkg/iris/eventsEcst.go

```diff
@@ -88,3 +105,103 @@ func (cb *commandBuilder) Build() models.CommandEvent {
 		Body:       *body,
 	}
 }
+
+// CreateEcstDiscoveryEvent ECST Discovery Items
+func CreateEcstDiscoveryEvent(eventType string, changeAction string, data models.Data, runId string, workspaceId string, configId string) models.DiscoveryEvent {
+	// Metadata for the event
+	id := GenerateId(workspaceId, configId, data)
+	time := time2.Now().Format(time2.RFC3339)
+
+	var header models.HeaderProperties
+	if eventType == EventTypeChange {
+		header = models.HeaderProperties{
+			Id:     id,
+			Scope:  fmt.Sprintf(EventScopeFormat, workspaceId, configId),
+			Class:  EventClass,
+			Type:   eventType,
+			Action: changeAction,
+		}
+	} else {
+		header = models.HeaderProperties{
+			Id:    id,
+			Scope: fmt.Sprintf(EventScopeFormat, workspaceId, configId),
+			Class: EventClass,
+			Type:  eventType,
+		}
+	}
+
+	body := models.DiscoveryBody{
+		State: models.State{
+			Name:   data.Cluster.Namespace,
+			Source: fmt.Sprintf("kubernetes/%s#%s", data.Cluster.Name, runId),
+			Time:   time,
+			Data:   data,
+		},
+	}
+
+	// Build ECST event
+	ecstDiscoveryEvent := NewEcstBuilder().
+		Header(header).
+		Body(body).
+		Build()
+	return ecstDiscoveryEvent
+}
+
+func GenerateId(workspaceId string, configId string, data models.Data) string {
+	scope := fmt.Sprintf(EventScopeFormat, workspaceId, configId)
+	// workspace/{workspaceId}/configuration/{configurationId}/discoveryItem/service/kubernetes/{clusterName}/{namespaceName}
+	idString := fmt.Sprintf("%s/%s/%s/%s", scope, EventClass, data.Cluster.Name, data.Cluster.Namespace)
+	sum := sha256.Sum256([]byte(idString))
+	id := hex.EncodeToString(sum[:])
+	return id
+}
+
+// Command Events
+func CreateStartReplay(workspaceId string, config kubernetesConfig) models.CommandEvent {
+	// Metadata for the command event
+	eventType := fmt.Sprintf("command")
+	action := fmt.Sprintf("startReplay")
+	scope := fmt.Sprintf(EventScopeFormat, workspaceId, config.ID)
+	header := models.CommandProperties{
+		Type:   eventType,
+		Action: action,
+		Scope:  scope,
+	}
+
+	startReplayEvent := NewCommand().Header(header).Build()
+	return startReplayEvent
+}
+
+func CreateEndReplay(workspaceId string, config kubernetesConfig) models.CommandEvent {
+	// Metadata for the command event
+	eventType := fmt.Sprintf("command")
+	action := fmt.Sprintf("endReplay")
+	scope := fmt.Sprintf(EventScopeFormat, workspaceId, config.ID)
+	header := models.CommandProperties{
+		Type:   eventType,
+		Action: action,
+		Scope:  scope,
+	}
+
+	endReplayEvent := NewCommand().Header(header).Build()
+	return endReplayEvent
+}
+
+func CreateDiscoveryItem(cluster models.Cluster, source string, scope string) models.DiscoveryItem {
```

Comment:
This is legacy discovery items right? If so, I would  take this out of "eventsEcst.go" so we can easily pinpoint the old stuff later

Reply:
Done